src/tests/ftest/dfuse/daos_build.py

```diff
@@ -49,45 +52,90 @@ def test_daos_build(self):
+
+        for key, value in cont_attrs.items():
+            daos_cmd.container_set_attr(pool=self.pool.uuid, cont=self.container.uuid,
+                                        attr=key, val=value)
 
         self.start_dfuse(self.hostlist_clients, self.pool, self.container)
 
         mount_dir = self.dfuse.mount_dir.value
         build_dir = os.path.join(mount_dir, 'daos')
 
-        cmds = ['git clone https://github.com/daos-stack/daos.git {}'.format(build_dir),
+        intercept = self.params.get('use_intercept', '/run/intercept/*', default=False)
+
+        remote_env = OrderedDict()
+        remote_env['PATH'] = '{}:$PATH'.format(os.path.join(mount_dir, 'venv', 'bin'))
+        remote_env['VIRTUAL_ENV'] = os.path.join(mount_dir, 'venv')
+
+        if intercept:
+            remote_env['LD_PRELOAD'] = os.path.join(self.prefix, 'lib64', 'libioil.so')
+            remote_env['D_LOG_FILE'] = '/var/tmp/daos_testing/daos-il.log'
+            remote_env['DD_MASK'] = 'all'
+            remote_env['DD_SUBSYS'] = 'all'
+            remote_env['D_LOG_MASK'] = 'INFO,IL=DEBUG'
+
+        envs = ['export {}={}'.format(env, value) for env, value in remote_env.items()]
+
+        preload_cmd = ';'.join(envs)
+
+        cmds = ['python3 -m venv {}/venv'.format(mount_dir),
+                'git clone https://github.com/daos-stack/daos.git {}'.format(build_dir),
                 'git -C {} submodule init'.format(build_dir),
                 'git -C {} submodule update'.format(build_dir),
-                '{} -C {} --jobs 50 build --build-deps=yes'.format(scons, build_dir)]
+                'python3 -m pip install pip --upgrade',
+                'python3 -m pip install -r {}/requirements.txt'.format(build_dir),
+                'scons -C {} --jobs 50 build --build-deps=yes'.format(build_dir)]
         for cmd in cmds:
             try:
-                # Set a timeout of 1500 seconds, the whole test will timeout after 1800
-                ret_code = general_utils.pcmd(self.hostlist_clients, cmd, timeout=1500)
+                command = '{};{}'.format(preload_cmd, cmd)
+                # Use a 10 minute timeout for most commands, but vary the build timeout based on
+                # the dfuse mode.
+                timeout = 10 * 60
+                if cmd.startswith('scons'):
+                    timeout = build_time * 60
+                ret_code = general_utils.pcmd(self.hostlist_clients, command, timeout=timeout)
                 if 0 in ret_code:
                     continue
                 self.log.info(ret_code)
                 raise CommandFailure("Error running '{}'".format(cmd))
             except CommandFailure as error:
-                self.log.error("BuildDaos Test Failed: %s", str(error))
-                self.fail("Test was expected to pass but it failed.\n")
+                self.log.error('BuildDaos Test Failed: %s', str(error))
+                self.fail('Unable to build daos over dfuse in mode {}.\n'.format(cache_mode))
```

Comment:
(pylint-consider-using-f-string) Formatting a regular string which could be a f-string